### PR TITLE
fix(nextjs): Warn about misplaced middleware file when using auth or getAuth

### PR DESCRIPTION
**Pull request by nikosdouvlis**
For NextJS 13 application, if the `/src` directory is used then the middleware file needs to be nested under it, otherwise the middleware will not run.

This is a very common user error - this commit updates the error message to list moving middleware under /src as a potential fix.

Fixes SDK-818

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/nextjs/src/server/errors.ts [hunk 1]
```diff
@@ -29,10 +29,20 @@ Alternatively, you can set your own ignoredRoutes. See https://clerk.com/docs/ne
 `;
 
 export const getAuthAuthHeaderMissing = () =>
-  'You need to use "authMiddleware" (or the deprecated "withClerkMiddleware") in your Next.js middleware file. You also need to make sure that your middleware matcher is configured correctly and matches this route or page. See https://clerk.com/docs/quickstarts/get-started-with-nextjs';
+  `Clerk: auth() was called but it looks like you aren't using authMiddleware in your middleware file. Please ensure the following:
+- Use "authMiddleware" (or the deprecated "withClerkMiddleware").
```
Comment (SokratisVidros) [inline packages/nextjs/src/server/errors.ts]: It's time to remove the deprecated proposal completely.

File: packages/nextjs/src/server/errors.ts [hunk 2]
```diff
@@ -29,10 +29,20 @@ Alternatively, you can set your own ignoredRoutes. See https://clerk.com/docs/ne
...
+  `Clerk: getAuth() was called but it looks like you aren't using authMiddleware in your middleware file. Please ensure the following:
+- Use "authMiddleware" (or the deprecated "withClerkMiddleware") in your middleware file.
+- Make sure your middleware matcher is configured correctly and it matches this route or page.
+- If you are using the /src directory, ensure that the middleware.ts file is located under it.
+For more details, see https://clerk.com/docs/quickstarts/get-started-with-nextjs.
+`;
 
 export const authAuthHeaderMissing = () =>
-  "Clerk: auth() was called but it looks like you aren't using `authMiddleware` in your middleware file. Please use `authMiddleware` and make sure your middleware matcher is configured correctly and it matches this route or page. See https://clerk.com/docs/quickstarts/get-started-with-nextjs";
+  `Clerk: auth() was called but it looks like you aren't using authMiddleware in your middleware file. Please ensure the following:
```
Comment (SokratisVidros) [inline packages/nextjs/src/server/errors.ts:40]: Please make the `auth()` `getAuth()` part dynamic and consolidate these two messages to save the extra bytes

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: packages/nextjs/src/server/errors.ts (hunk 1 above)
Comment (nikosdouvlis) [reply packages/nextjs/src/server/errors.ts]: Done

File: packages/nextjs/src/server/errors.ts (hunk 2 above)
Comment (nikosdouvlis) [reply packages/nextjs/src/server/errors.ts:40]: Done

**Review by SokratisVidros — APPROVED** [review]

File: packages/nextjs/src/server/errors.ts [hunk 3]
```diff
@@ -28,11 +28,15 @@ Alternatively, you can set your own ignoredRoutes. See https://clerk.com/docs/ne
...
-export const getAuthAuthHeaderMissing = () =>
-  'You need to use "authMiddleware" (or the deprecated "withClerkMiddleware") in your Next.js middleware file. You also need to make sure that your middleware matcher is configured correctly and matches this route or page. See https://clerk.com/docs/quickstarts/get-started-with-nextjs';
-
-export const authAuthHeaderMissing = () =>
-  "Clerk: auth() was called but it looks like you aren't using `authMiddleware` in your middleware file. Please use `authMiddleware` and make sure your middleware matcher is configured correctly and it matches this route or page. See https://clerk.com/docs/quickstarts/get-started-with-nextjs";
+export const getAuthAuthHeaderMissing = () => authAuthHeaderMissing('getAuth');
+
+export const authAuthHeaderMissing = (helperName = 'auth') =>
+  `Clerk: ${helperName}() was called but it looks like you aren't using authMiddleware in your middleware file. Please ensure the following:
+- Use authMiddleware in your middleware file.
```
Comment (SokratisVidros) [inline packages/nextjs/src/server/errors.ts:35]:
```suggestion
- Use authMiddleware in your Next.js Middleware file.
```

Comment (nikosdouvlis) [reply packages/nextjs/src/server/errors.ts:35]: done - dropping file from the first sentence as this is how it's used in the NextJS docs: https://nextjs.org/docs/app/building-your-application/routing/middleware

File: packages/nextjs/src/server/errors.ts [hunk 4]
```diff
@@ -28,11 +28,15 @@ Alternatively, you can set your own ignoredRoutes. See https://clerk.com/docs/ne
...
+- Make sure your middleware matcher is configured correctly and it matches this route or page.
+- If you are using the /src directory, ensure that the middleware.ts file is located under it.
+For more details, see https://clerk.com/docs/quickstarts/get-started-with-nextjs.
```
Comment (BRKalow) [inline packages/nextjs/src/server/errors.ts:38]:
```suggestion
export const authAuthHeaderMissing = (helperName = 'auth') =>
  `Clerk: ${helperName}() was called but Clerk can't detect usage of authMiddleware(). Please ensure the following:
- authMiddleware() is used in your Next.js Middleware.
- Your Middleware matcher is configured correctly and it matches this route or page.
- If you are using the src directory, make sure the Middleware file is inside of it.

For more details, see https://clerk.com/docs/quickstarts/get-started-with-nextjs.
```

**Review by anagstef — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**